### PR TITLE
Fix image stuttering.

### PR DESCRIPTION
**Pull request by caiocotts**
Fix AlbumGrid shaking when a non-square album cover is rendered.

### CODE REVIEW

File: server/subsonic/media_retrieval.go [hunk 1]
```diff
@@ -64,8 +64,9 @@ func (api *Router) GetCoverArt(w http.ResponseWriter, r *http.Request) (*respons
 	p := req.Params(r)
 	id, _ := p.String("id")
 	size := p.IntOr("size", 0)
+	square := p.BoolOr("square", false)
```
Comment (dweymouth) [inline server/subsonic/media_retrieval.go:67]: `"square"` is not a parameter for Subsonic or OpenSubsonic getCoverArt. Probably can just pass false directly

Comment (dweymouth) [reply server/subsonic/media_retrieval.go:67]: Oh... looks like the web UI uses the subsonic api to fetch images?

Comment (deluan) [reply server/subsonic/media_retrieval.go:67]: Yeah, I proposed this. This is a quick "hack" to fix an issue in the (current) WebUI. 
I don't mean to document and/or propose this to OpenSubsonic, it is just an internal thing.

For the new UI We will introduce a proper native artwork endpoint, and this parameter will be removed from the `getArtWork` endpoint.

File: core/artwork/reader_resized.go [hunk 2]
```diff
@@ -90,15 +93,27 @@ func resizeImage(reader io.Reader, size int) (io.Reader, int, error) {
...
+	if originalSize >= size {
+		resized = imaging.Fit(original, size, size, imaging.Lanczos)
+	} else {
+		if bounds.Max.Y < bounds.Max.X {
+			resized = imaging.Resize(original, size, 0, imaging.Lanczos)
+		} else {
+			resized = imaging.Resize(original, 0, size, imaging.Lanczos)
+		}
+	}
+	if square {
```
Comment (deluan) [inline core/artwork/reader_resized.go:110]: We need test coverage for the new code paths

**Review by deluan — CHANGES_REQUESTED** [review]
Looking good, please see my comments below

File: core/artwork/artwork_internal_test.go [hunk 3]
```diff
@@ -211,27 +215,86 @@ var _ = Describe("Artwork", func() {
...
+	case "png":
+		f, _ := os.Create(filepath.Join(tmpDir, "cover.png"))
+		defer f.Close()
+		_ = png.Encode(f, img)
+	case "jpg":
+		f, _ := os.Create(filepath.Join(tmpDir, "cover.jpg"))
+		defer f.Close()
+		_ = jpeg.Encode(f, img, &jpeg.Options{Quality: 75})
+	}
+	// save variable img to variable f
```
Comment (deluan) [inline core/artwork/artwork_internal_test.go:297]: This comment does not seem to be in the right place, maybe it is not even needed?

File: core/artwork/artwork_internal_test.go [hunk 4]
```diff
@@ -211,27 +215,86 @@ var _ = Describe("Artwork", func() {
...
+	switch format {
+	case "png":
+		f, _ := os.Create(filepath.Join(tmpDir, "cover.png"))
+		defer f.Close()
+		_ = png.Encode(f, img)
+	case "jpg":
+		f, _ := os.Create(filepath.Join(tmpDir, "cover.jpg"))
+		defer f.Close()
+		_ = jpeg.Encode(f, img, &jpeg.Options{Quality: 75})
+	}
```
Comment (deluan) [inline core/artwork/artwork_internal_test.go:296]: Maybe simplify it like this:
```go
	tmpDir := GinkgoT().TempDir()
        f, _ := os.Create(filepath.Join(tmpDir, "cover." + format))
        defer f.Close()
	switch format {
	case "png":
		_ = png.Encode(f, img)
	case "jpg":
		_ = jpeg.Encode(f, img, &jpeg.Options{Quality: 75})
	}
```

**Review by deluan — CHANGES_REQUESTED** [review]
Nice tests. A couple of nitpicks and we should be good to merge

File: core/artwork/artwork_internal_test.go (hunk 3 above)
Comment (caiocotts) [reply core/artwork/artwork_internal_test.go:297]: Was trying to see what copilot would do and forgot to remove it. 
Oops.

**Pull request merged**